### DISCUSSION TITLE
Disable NX reading .env files behaviour from the nx.json config?

### DISCUSSION THREAD

**Discussion by RIP21** [nrwl/nx#27435] [Feature Requests]
This behavior
https://nx.dev/recipes/tips-n-tricks/define-environment-variables
It is a major pain as it is only controlled by an environment variable and I cannot easily disable it for my editor and all my team, aside from using `.envrc` e.g. `direnv` tool which complicates the initial setup for no good reason + also not really working well with IDEs (JetBrains is lacking good support for it if you execute commands from UI (Run configurations) and not the terminal, I assume VSCode isn't that better in that regard TBH, especially with NX extension). 

I'm using `dotenvx` for all my `.env` files with encryption functionality https://dotenvx.com/docs/quickstart#add-encryption which NX doesn't understand, so it conflicts with `dotenvx` reading `.env.*` files as it reads them before in encrypted form, and then `dotenvx` ignore these, as they're already available. 

I don't want to use `--override` mode of `direnvx`, as it just complicates things even more as I still wish for global variables to have precedence over those in `.env.local`.

I use NX in monorepo mode (not integrated).

**Comment by motdotla** [comment]
@RIP21 are you using dotenvx by requiring/importing it in code or by using the cli approach?

I *think* the cli approach might pre-empt nx's behavior here:

```
{
  "scripts": {
    "dev": "dotenvx run -- nxcommandhere"
  }
}
```

Have you given that a shot already?

**Reply by RIP21** [reply]
I need to get back to code to check 100%. But AFAIR I'm using CLI as it's the nuclear method, but it's not as nuclear as NX as it comes before.
If dotenvx would call NX, then it would nuke NX env reading, but if NX calls dotenvx then dotenvx gets nuked as whatever comes from export ENV wins over what dotenvx reads from env files.
